pkg/operator/controller/ingress/status_test.go

```diff
@@ -2239,3 +2338,114 @@ func TestComputeIngressUpgradeableCondition(t *testing.T) {
 		})
 	}
 }
+
+func TestComputeIngressEvaluationConditionsDetectedCondition(t *testing.T) {
+	const (
+		ingressDomain = "apps.foo.com"
+	)
+	testCases := []struct {
+		description string
+		mutate      func(*corev1.Service)
+		secret      *corev1.Secret
+		expect      bool
+	}{
+		{
+			description: "if the service.beta.kubernetes.io/load-balancer-source-ranges is unset",
+			mutate: func(svc *corev1.Service) {
+				svc.Annotations[LoadBalancerSourceRangesAnnotation] = ""
+			},
+			expect: false,
+		},
+		{
+			description: "if the service.beta.kubernetes.io/load-balancer-source-ranges is set",
+			mutate: func(svc *corev1.Service) {
+				svc.Annotations[LoadBalancerSourceRangesAnnotation] = "127.0.0.0/8"
+			},
+			expect: true,
+		},
+		{
+			description: "if the LoadBalancerSourceRanges and AllowedSourceRanges are empty",
+			mutate: func(svc *corev1.Service) {
+				svc.Spec.LoadBalancerSourceRanges = []string{}
+			},
+			expect: false,
+		},
+		{
+			description: "if the LoadBalancerSourceRanges is set when AllowedSourceRanges is empty",
+			mutate: func(svc *corev1.Service) {
+				svc.Spec.LoadBalancerSourceRanges = []string{"127.0.0.0/8"}
+			},
+			expect: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			ic := &operatorv1.IngressController{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+				Spec: operatorv1.IngressControllerSpec{
+					EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+						Type: operatorv1.LoadBalancerServiceStrategyType,
+						LoadBalancer: &operatorv1.LoadBalancerStrategy{
+							AllowedSourceRanges: []operatorv1.CIDR{},
+						},
+					},
+				},
+				Status: operatorv1.IngressControllerStatus{
+					EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+						Type: operatorv1.LoadBalancerServiceStrategyType,
+					},
+					Domain: ingressDomain,
+				},
+			}
+			trueVar := true
+			deploymentRef := metav1.OwnerReference{
+				APIVersion: "apps/v1",
+				Kind:       "Deployment",
+				Name:       "router-default",
+				UID:        "1",
+				Controller: &trueVar,
+			}
+			platformStatus := &configv1.PlatformStatus{
+				Type: configv1.AWSPlatformType,
+				AWS: &configv1.AWSPlatformStatus{
+					ResourceTags: []configv1.AWSResourceTag{
+						{
+							Key:   "Key1",
+							Value: "Value1",
+						},
+					},
+				},
+			}
+			currentService := &corev1.Service{
+				Spec: corev1.ServiceSpec{
+					LoadBalancerSourceRanges: []string{},
+				},
+			}
+
+			wantSvc, service, err := desiredLoadBalancerService(ic, deploymentRef, platformStatus, currentService)
+			if err != nil {
+				t.Errorf("%q: unexpected error from desiredLoadBalancerService: %v", tc.description, err)
+				return
+			}
+			if !wantSvc {
+				t.Errorf("%q: unexpected false value from desiredLoadBalancerService", tc.description)
+				return
```

Comment:
As this is wrapped using `t.Run`, there is no need to repeat the description, and is there any reason not to use `t.Fatalf`/`t.Fatal`?
```suggestion
				t.Fatalf("unexpected error from desiredLoadBalancerService: %v", err)
			}
			if !wantSvc {
				t.Fatal("unexpected false value from desiredLoadBalancerService")
```

Reply:
No reason. I applied this.